java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -649,8 +667,17 @@ public final ColumnVector ifElse(Scalar trueValue, Scalar falseValue) {
   public final ColumnVector[] slice(int... indices) {
     long[] nativeHandles = slice(this.getNativeView(), indices);
     ColumnVector[] columnVectors = new ColumnVector[nativeHandles.length];
-    for (int i = 0; i < nativeHandles.length; i++) {
-      columnVectors[i] = new ColumnVector(nativeHandles[i]);
+    try {
+      for (int i = 0; i < nativeHandles.length; i++) {
+        columnVectors[i] = new ColumnVector(nativeHandles[i]);
+      }
+    } catch (Throwable t) {
+      for (ColumnView columnView: columnVectors) {
```

Comment:
In addition to closing the column vectors/views in the array, we also have pending `nativeHandles`, sorry I just caught this one. In order to really make safer, we should also need to close the handles which are leaked. Should probably create a helper handles -> vectors, or handles -> views, if we don't have those already, since all the other places you changed are similar.

I would change the loop like so:

```
try {
  for (int i = 0; i < nativeHandles.length; i++) {
    columnVectors[i] = new ColumnVector(nativeHandles[i]);
    nativeHandles[i] = 0; // mark the handle as consumed
  }
} catch (Throwable t) {
  for (ColumnVector cv : columnVectors) {
    if (cv != null) {
      cv.close();
    }
  }
  for (long handle : nativeHandles) { 
    if (handle != 0) {
      ColumnVector.deleteCudfColumn(handle);
    }
  }
  throw t;
}
```

Reply:
Good catch! But why not close all the `nativeHandles`? We got them as a result of `slice` They should all be cleaned out since we have encountered an error.

Reply:
I believe the same thing should happen in `splitAsViews`

Reply:
Yes, but since you have handed some handles to the vector/view, the expectation is that the vector/view will close the handle they were given. So you don't want to double delete these.